vllm/worker/model_runner.py

```diff
@@ -264,6 +345,37 @@ def __post_init__(self):
             self.lora_index_mapping = [[] for _ in range(self.n_seqs)]
             self.lora_prompt_mapping = [[] for _ in range(self.n_seqs)]
 
+    @classmethod
+    def gen_inter_data_builder(cls, num_seqs: int):
+        return lambda: ModelInputForGPUBuilder.InterDataForSeqGroup(
+            request_id="",
+            seq_ids=[0] * num_seqs,
+            is_prompt=True,
+            block_tables=None,
+            computed_block_nums=[])
+
+    inter_data_cache: Dict[int, PyObjectCache] = {}
+
+    @classmethod
+    def init_cached_inter_data(cls, *args, **kwargs):
```

Comment:
same comment as above, I think this should be a per-instance cache. users might have multiple vLLM instance in the same process, e.g. offline inference.

Reply:
Refactored